src/Sarif.Viewer.VisualStudio.ResultSources.DeveloperCanvas.Core/Services/DevCanvasResultSourceService.cs

```diff
@@ -0,0 +1,384 @@
+        // Whenever a query completes, we process the next query on the queue until all queries are complete.
+        // We use a priority queue to ensure we query for visible/opened files before files that are just loaded in the background.
+        private int outstandingQueries = 0;
+
+        /// <summary>
+        /// Number of parallel queries we allow at once.
+        /// </summary>
+        private const int maxParallelQueries = 5;
+
+        /// <summary>
+        /// Max total number of entries into the cache that we allow.
+        /// </summary>
+        private const int MaxInsightCacheSize = 500;
+
+        private static readonly string MaxInsightErrorMessage = $"Insights were not queried for one or more files because the limit of {MaxInsightCacheSize} has been reached. You can adjust this limit in the Options menu.";
+
+        /// <summary>
+        /// The number of minutes to wait before evicting an entry from the cache.
+        /// </summary>
+        private const int minutesBeforeRefresh = 60;
+
+        private readonly Queue<string> filePathQueue;
+        private readonly ReaderWriterLockSlim filePathQueueLock;
+
+        /// <summary>
+        /// Handles the actual accessing of data 
+        /// </summary>
+        private readonly DevCanvasAccessor accessor;
+
+        public DevCanvasResultSourceService(
+            string solutionRootPath,
+            Func<string, bool> getOptionStateCallback,
+            IServiceProvider serviceProvider,
+            IHttpClientAdapter httpClientAdapter,
+            ISecretStoreRepository secretStoreRepository,
+            IFileWatcher fileWatcherGitPush,
+            IFileWatcher fileWatcherBranchChange,
+            IFileSystem fileSystem,
+            IGitExe gitExe,
+            IInfoBarService infoBarService,
+            IStatusBarService statusBarService)
+        {
+            this.GetOptionStateCallback = getOptionStateCallback;
+            this.serviceProvider = serviceProvider;
+            this.httpClientAdapter = httpClientAdapter;
+            this.secretStoreRepository = secretStoreRepository;
+            this.fileWatcherBranchChange = fileWatcherBranchChange;
+            this.fileWatcherGitPush = fileWatcherGitPush;
+            this.fileSystem = fileSystem;
+            this.gitExe = gitExe;
+            this.gitExe.RepoPath = solutionRootPath;
+            this.infoBarService = infoBarService;
+            this.statusBarService = statusBarService;
+
+            this.browserService = new BrowserService();
+            this.filesQueriedCache = new MemoryCache("FilesQueried");
+
+            filePathQueue = new Queue<string>();
+
+            accessor = new DevCanvasAccessor();
+        }
+
+        /// <inheritdoc/>
+        public async System.Threading.Tasks.Task InitializeAsync()
+        {
+            string repoPath = await gitExe.GetRepoRootAsync();
```

Comment:
How is getting the repoPath and repoRoot useful here?  Can't any file that is seen be from anywhere?  How does this know what context is relevant?